### ISSUE TITLE
Variable is read but never assigned not triggered when variable assigned /after/ read

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#628]
The 'variable is read but never assigned' rule does not detect the following smell:
````
DEFINE VARIABLE foo AS LOGICAL NO-UNDO.

IF foo THEN
   MESSAGE "bar":u.
ELSE
   foo = TRUE.
````
When the assignment to foo is removed it works fine. 

The rule **should** be checking that the first use of the variable is **after** the first assign of the variable.

**Comment by gquerret** [comment]
Agree it should be detected, but this level of granularity is not available yet. We'd need to have better flow control to detect this case.

**Comment by gquerret** [comment]
Code flow information is now available in the parse tree, so enhancement can now be implemented.
